src/decision-engine/index.js

```diff
@@ -90,35 +81,25 @@ class DecisionEngine {
     const uniqCids = uniqWith((a, b) => a.equals(b), cids)
     const groupedTasks = groupBy(task => task.target.toB58String(), tasks)
 
-    waterfall([
-      (callback) => map(uniqCids, (cid, cb) => {
-        this.blockstore.get(cid, cb)
-      }, callback),
-      (blocks, callback) => each(Object.values(groupedTasks), (tasks, cb) => {
-        // all tasks have the same target
-        const peer = tasks[0].target
-        const blockList = cids.map((cid) => {
-          return blocks.find(b => b.cid.equals(cid))
-        })
-
-        this._sendBlocks(peer, blockList, (err) => {
-          if (err) {
-            // `_sendBlocks` actually doesn't return any errors
-            this._log.error('should never happen: ', err)
-          } else {
-            blockList.forEach((block) => this.messageSent(peer, block))
-          }
-
-          cb()
-        })
-      }, callback)
-    ], (err) => {
-      this._tasks = []
-
-      if (err) {
-        this._log.error(err)
+    const blocks = await Promise.all(uniqCids.map(cid => this.blockstore.get(cid)))
+    await Object.values(groupedTasks).map(async (tasks) => {
```

Comment:
This doesnt need a Promise.all to be awaited?

Reply:
`this.blockstore.get(cid)` returns a promise, if that's what you mean?

Reply:
No Promise.all(Object.values().map() )

Reply:
😬